pkg/resources/kafka/headlessService.go

```diff
@@ -92,3 +97,96 @@ func (r *Reconciler) deleteHeadlessService() error {
 
 	return err
 }
+
+// deleteNonHeadlessServices deletes the all-broker service that was created for the current KafkaCluster
+// if there is any and also the service of each broker
+func (r *Reconciler) deleteNonHeadlessServices(ctx context.Context) error {
+	svc := corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.KafkaCluster.GetNamespace(),
+			Name:      fmt.Sprintf(kafkautils.AllBrokerServiceTemplate, r.KafkaCluster.GetName()),
+		},
+	}
+
+	err := r.Client.Delete(ctx, &svc)
+	if err != nil && client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	// delete broker services
+	labelSelector := labels.NewSelector()
+	for k, v := range apiutil.LabelsForKafka(r.KafkaCluster.GetName()) {
+		req, err := labels.NewRequirement(k, selection.Equals, []string{v})
+		if err != nil {
+			return err
+		}
+		labelSelector = labelSelector.Add(*req)
+	}
+
+	// add "has label 'brokerId' to matching labels selector expression
+	req, err := labels.NewRequirement(v1beta1.BrokerIdLabelKey, selection.Exists, nil)
+	if err != nil {
+		return err
+	}
+	labelSelector = labelSelector.Add(*req)
+
+	var services corev1.ServiceList
+	err = r.Client.List(ctx, &services,
+		client.InNamespace(r.KafkaCluster.GetNamespace()),
+		client.MatchingLabelsSelector{Selector: labelSelector},
+	)
+
+	if err != nil {
+		return errors.WrapIfWithDetails(err, "failed to list services",
+			"namespace", r.KafkaCluster.GetNamespace(),
+			"label selector", labelSelector.String())
+	}
+
+	// if NodePort is used for any of the external listeners, the corresponding services need to remain
+	// so that clients from outside the Kubernetes cluster can reach the brokers
+	filteredSvcsToDelete := services
+	if r.checkIfNodePortSvcNeeded() {
+		filteredSvcsToDelete = getNonNodePortSvc(services)
+	}
+
+	for i := range filteredSvcsToDelete.Items {
+		svc = filteredSvcsToDelete.Items[i]
+		if !svc.GetDeletionTimestamp().IsZero() {
+			continue
+		}
+		err = r.Client.Delete(ctx, &svc)
+		if err != nil && client.IgnoreNotFound(err) != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// checkIfNodePortSvcNeeded returns true when users specify any of the external listeners to use NodePort
+func (r *Reconciler) checkIfNodePortSvcNeeded() bool {
+	for _, externalListener := range r.KafkaCluster.Spec.ListenersConfig.ExternalListeners {
+		if externalListener.GetAccessMethod() == corev1.ServiceTypeNodePort {
+			return true
+		}
+	}
+	return false
+}
+
+func getNonNodePortSvc(services corev1.ServiceList) corev1.ServiceList {
+	var svc corev1.Service
+	var nonNodePortSvc corev1.ServiceList
```

Comment:
**Recommended**: We **SHOULD** consider calling this plural because it refers to a list of services - or alternatively call it a list, but I think that is too verbose and unnecessary, it is a convenience that we use a list struct and not the slice directly - because currently the name itself would imply this designates a single service while that is not the case.

Apologies I missed this on the first review, only caught it while reviewing my comments.